WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -339,23 +340,17 @@ private void addCommentApproveActionForCommentNotification(NotificationCompat.Bu
         // adding comment approve action
         Intent commentApproveIntent = getCommentActionIntent();
         commentApproveIntent.addCategory(KEY_CATEGORY_COMMENT_MODERATE);
-        commentApproveIntent.putExtra(NotificationsListFragment.NOTE_INSTANT_APPROVE_EXTRA, true);
+        commentApproveIntent.putExtra(NotificationsProcessingService.ARG_ACTION_TYPE, NotificationsProcessingService.ARG_ACTION_APPROVE);
         if (noteId != null) {
-            commentApproveIntent.putExtra(NotificationsListFragment.NOTE_ID_EXTRA, noteId);
+            commentApproveIntent.putExtra(NotificationsProcessingService.ARG_NOTE_ID, noteId);
         }
-        PendingIntent commentApprovePendingIntent = PendingIntent.getActivity(this, 0, commentApproveIntent,
-                PendingIntent.FLAG_CANCEL_CURRENT);
+        PendingIntent commentApprovePendingIntent =  PendingIntent.getService(this, 0, commentApproveIntent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
ok fixed this in this commit https://github.com/wordpress-mobile/WordPress-Android/pull/4667/commits/3c7a7a434aae136476abc8b4ae56dcba14be00ec

+ changed the UPDATE for the CANCEL flag
+ added a thread per each call to the service so we can make sure these run in parallel
However, at this very point in time there is very little possibility to have 2 quick actions to be executed at the same time, because if we have 2 pending notifications these are grouped together (so no quick actions are shown at all), and in the case 1 unique notification is shown, showing the REPLY and the LIKE/APPROVE actions at the same time, for instance, the only time when this could potentially happen is the case that a user would hit REPLY, write the reply in the direct reply field, and then LIKE rapidly right after hitting SEND. Anyway, if it happens, it’s handled by this threading mechanism now.
I've tested this by adding a `sleep` call in line 155 in `NotificationsProcessingService.java`, here https://github.com/wordpress-mobile/WordPress-Android/blob/issue/4659-notifications-service/WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java#L154 (it's of course a blank line now)
+ I also evaluated  using `IntentService` instead of `Service`, but I preferred to handle actions in parallel using threads as to keep this available for the future (IntentService would queue intents and that might not be what we really want as a UX behavior.)